_posts/2017-10-28-soap-to-json-converter.md

```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+title: Creating a SOAP to JSON converter service
+date: 2017-10-28
+---
+
+I recently had to consume a SOAP based web service from within JavaScript code. To solve that problem I decided to write a separate service acting as a SOAP to JSON converter. In this post I explain why and how I did it.
+
+#### The problem
+
+I am currently writing a web application to display train data and one of the data source I am using is a SOAP based web service. SOAP stands for Simple Object Access Protocol, it is a protocol to exchange data over [HTTP](https://en.wikipedia.org/wiki/Hypertext_Transfer_Protocol) using [XML](https://en.wikipedia.org/wiki/XML) as message format.
+SOAP can be very powerful but is not as easy to use out of the box as is JSON using JavaScript.
```

Comment:
This sentence is a bit clunky 

Do you mean that it's "not as easy to use out of the box as JSON in JavaScript"?

Reply:
oh yeah, that's awful, rephrasing...